crashed when minimize a window which contains a stroked rectangle

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

panic: runtime error: index out of range [0] with length 0

goroutine 18 [running, locked to thread]:
golang.org/x/image/vector.(*Rasterizer).rasterizeDstRGBASrcUniformOpOver(0xc0000301a0, 0xc0000aa0c0, {{0xc008000000000000?, 0xc008000000000000?}, {0x3ff0000000000000?, 0x0?}}, 0xffff, 0xffff, 0xffff, 0xffff)
	C:/Users/Administrator/go/pkg/mod/golang.org/x/image@v0.0.0-20220601225756-64ec528b34cd/vector/vector.go:398 +0x305
golang.org/x/image/vector.(*Rasterizer).Draw(0xc0000301a0, {0x1718bc8?, 0xc0000aa0c0}, {{0x3ff0000000000000?, 0x0?}, {0x0?, 0x3ff0000000000000?}}, {0x1718220?, 0xc0000aa100?}, {0x0, ...})
	C:/Users/Administrator/go/pkg/mod/golang.org/x/image@v0.0.0-20220601225756-64ec528b34cd/vector/vector.go:287 +0x1ab
github.com/srwiley/rasterx.(*ScannerGV).Draw(0xc0000301a0)
	C:/Users/Administrator/go/pkg/mod/github.com/srwiley/rasterx@v0.0.0-20210519020934-456a8d69b780/scan.go:141 +0x85
fyne.io/fyne/v2/internal/painter.DrawRectangle(0xc0004001e0, 0x40400000, 0xc000591730)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/draw.go:102 +0x310
fyne.io/fyne/v2/internal/painter/gl.(*painter).newGlStrokedRectTexture(0xc0000aa080, {0x171a5a0?, 0xc0004001e0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/texture.go:156 +0xa9
fyne.io/fyne/v2/internal/painter/gl.(*painter).newGlRectTexture(0x15bd0c0?, {0x171a5a0?, 0xc0004001e0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/texture.go:146 +0x4e
fyne.io/fyne/v2/internal/painter/gl.(*painter).getTexture(0xc0000aa080, {0x171a5a0, 0xc0004001e0}, 0xc0005918f8)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/texture.go:36 +0x4f
fyne.io/fyne/v2/internal/painter/gl.(*painter).drawTextureWithDetails(0xc0000aa080, {0x171a5a0?, 0xc0004001e0?}, 0xc0002e6180?, {0x5918b0?, 0xc0?}, {0xc73d29?, 0x0?}, {0xf2de62?, 0x0?}, ...)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/draw.go:134 +0x8b
fyne.io/fyne/v2/internal/painter/gl.(*painter).drawRectangle(0xc0000aa080, 0xc0004001e0, {0xc7e165?, 0x0?}, {0x171a9c0?, 0x0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/draw.go:104 +0x185
fyne.io/fyne/v2/internal/painter/gl.(*painter).drawObject(0xc0005919a0?, {0x171a5a0?, 0xc0004001e0?}, {0xc7bedd?, 0x0?}, {0x1575ee0?, 0x0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/draw.go:86 +0x1c5
fyne.io/fyne/v2/internal/painter/gl.(*painter).Paint(0xceb2c5?, {0x171a5a0, 0xc0004001e0}, {0x40c9c0?, 0xc0?}, {0xcca3b0?, 0xc0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/gl/painter.go:84 +0x74
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).paint.func1(0xecc011?, {0xecc69c?, 0x0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/canvas.go:284 +0xff
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree.func1({0x171a5a0?, 0xc0004001e0}, {0x4001e0?, 0xc0?}, {0x7a6320?, 0xc0?}, {0x171a600?, 0x0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/common/canvas.go:438 +0x277
fyne.io/fyne/v2/internal/driver.walkObjectTree({0x171a5a0, 0xc0004001e0}, 0x0, {0x171a300, 0xc0007a6320}, {0x15924c0?, 0x0?}, {0xcca460?, 0xc0?}, {0x7a1ec0?, ...}, ...)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/util.go:169 +0x29b
fyne.io/fyne/v2/internal/driver.walkObjectTree.func1(...)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/util.go:176
fyne.io/fyne/v2/internal/driver.walkObjectTree({0x171a300, 0xc0007a6320}, 0x0, {0x0, 0x0}, {0x0?, 0x0?}, {0x591c80?, 0xc0?}, {0xcb1771?, ...}, ...)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/util.go:190 +0x41d
fyne.io/fyne/v2/internal/driver.WalkVisibleObjectTree({0x171a300?, 0xc0007a6320?}, 0xc000591cb0?, 0xccec25?)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/util.go:134 +0x4c
fyne.io/fyne/v2/internal/driver/common.(*Canvas).walkTree(0xc000000000?, 0xc000089ee0, 0xc000591dd0, 0xc000591db8)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/common/canvas.go:459 +0x165
fyne.io/fyne/v2/internal/driver/common.(*Canvas).WalkTrees(0xc000422000, 0xc000591da0?, 0x11a4886?)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/common/canvas.go:371 +0x3c
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).paint(0xc000422000, {0x0?, 0x0?})
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/canvas.go:296 +0xfe
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/loop.go:203 +0x8f
fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0x1182f06?, 0xc000591e80)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/window.go:903 +0x4f
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0xc000422000?, 0x0?)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/loop.go:194 +0x4a
fyne.io/fyne/v2/internal/driver/glfw.(*window).platformResize.func1()
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/window_notlinux.go:17 +0x48
fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0xc000591f90?, 0xc000532000)
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/window.go:903 +0x4f
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread.func1()
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/loop.go:234 +0x13b
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread
	C:/Users/Administrator/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/loop.go:226 +0xea
exit status 2

### How to reproduce

1. build a background rectangle and set it's stroke width and  color

```go
	bg := canvas.NewRectangle(color.White)
	bg.StrokeColor = color.Gray{128}
	bg.StrokeWidth = 1
```

2.  use MaxLayout to put the content above the rectangle
3.  run app and click the minimize button of  window

### Screenshots

![save_20230110_101431](https://user-images.githubusercontent.com/10773905/211446497-e72db9df-cb4b-45db-8fac-27c52b217114.jpg)


### Example code

```go
func main() {
	a := app.New()
	w := a.NewWindow("Hello")

	bg := canvas.NewRectangle(color.White)
	bg.StrokeColor = color.Gray{128}
	bg.StrokeWidth = 1

	hello := widget.NewLabel("Hello Fyne!")
	w.SetContent(container.NewMax(
		bg,
		container.NewVBox(
			hello,
			widget.NewButton("Hi!", func() {
				hello.SetText("Welcome :)")
			})),
	))

	w.ShowAndRun()
}
```

### Fyne version

2.3.0

### Go compiler version

1.18

### Operating system

Windows

### Operating system version

Windows7

### Additional Information

_No response_

--- a/internal/driver/glfw/canvas.go
+++ b/internal/driver/glfw/canvas.go
@@ -281,6 +281,9 @@ func (c *glCanvas) paint(size fyne.Size) {
 			inner := clips.Push(pos, obj.Size())
 			c.Painter().StartClipping(inner.Rect())
 		}
+		if size.Width <= 0 || size.Height <= 0 { // iconifying on Windows can do bad things
+			return
+		}
 		c.Painter().Paint(obj, pos, size)
 	}
 	afterPaint := func(node *common.RenderCacheNode) {

